src/deprecations.jl

```diff
@@ -7,3 +7,6 @@
 Base.@deprecate_binding DisjointSets DisjointSet
 Base.@deprecate_binding IntDisjointSets IntDisjointSet
 @deprecate DisjointSets(xs...) DisjointSet(xs)
+# Deprecations from #728
+@deprecate enqueue! Base.push!
+@deprecate dequeue! Base.popfirst!
```

Comment:
I think you need either 
`@deprecate enqueue!(x) push!(x)`
or
`@deprecate_binding enqueue! push!`

Also we should copy all the tests that use the deprecated method into the `test/deprecations.jl` file, or at least a resprentative sample of them, 
so that we can make sure the deprecation does still work until it is removed